tests/test_utils.py

```diff
@@ -220,6 +220,53 @@ def test_extract_image_from_dockerfile_from_with_build_args(self) -> None:
         # Should extract the variable reference
         self.assertEqual(result, '${BASE_IMAGE}')
 
+    def test_extract_image_from_dockerfile_with_resource_url(self) -> None:
+        """Test extracting Docker image using ResourceUrl scheme."""
+        dockerfile_content = """FROM alpine:3.18
+RUN apk add --no-cache python3
+"""
+        # Create in repository subdirectory
+        repo_dir = self.temp_path / 'repository'
+        repo_dir.mkdir()
+        dockerfile_path = repo_dir / 'Dockerfile'
+        dockerfile_path.write_text(dockerfile_content)
+
+        result = utils.extract_image_from_dockerfile(
+            self.context, models.ResourceUrl('repository:///Dockerfile')
+        )
+
+        self.assertEqual(result, 'alpine:3.18')
+
+    def test_extract_image_from_dockerfile_with_string_path(self) -> None:
+        """Test extracting Docker image with string path (not Path object)."""
+        dockerfile_content = """FROM node:18-alpine
+WORKDIR /usr/src/app
+"""
+        dockerfile_path = self.temp_path / 'Dockerfile.node'
+        dockerfile_path.write_text(dockerfile_content)
```

Comment:
Actually ... after looking at a few of these ... what about adding a helper function to write a temp file by name?
```python
class MyTest:
    def write_temporary_file(content: str) -> str:
        file_name = self.temp_path / str(uuid.uuid4())
        pathlib.Path(file_name).write_text(textwrap.dedent(content).lstrip('\n'))
        return file_name

    def test_extract_image_from_dockerfile_with_string_path(self) -> None:
        file_name = self.write_temporary_file("""
            FROM node:18-alpine
            WORKDIR /usr/src/app
        """)
        result = utils.extract_image_from_dockerfile(self.context, file_name)
        ...
````